code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -122,25 +126,29 @@
 			update_inv_glasses()
 		if(ITEM_SLOT_GLOVES)
 			if (gloves && swap)
-				gloves.dropped(src, TRUE)
 				current_equip = gloves
+				if (!temporarilyRemoveItemFromInventory(gloves))
+					return
 			gloves = I
 			update_inv_gloves()
 		if(ITEM_SLOT_FEET)
 			if (shoes && swap)
-				shoes.dropped(src, TRUE)
 				current_equip = shoes
+				if (!temporarilyRemoveItemFromInventory(shoes))
```

Comment:
From the proc def
```
//visibly unequips I but it is NOT MOVED AND REMAINS IN SRC
//item MUST BE FORCEMOVE'D OR QDEL'D
```
Have you considered this? Did I miss something in item code?

Reply:
Later on:
```
	if (current_equip)
		put_in_active_hand(current_equip)
```

...which I believe solves this problem.